File: backend/src/main/java/com/EduPay/dto/PaymentRequest.java
package com.EduPay.dto;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class PaymentRequest {
    private Long studentId;
    private Long feeId;
    private Double amount;
    private String currency;
    private String description; // Description for the payment gateway

}

File: lib/models/payment_request.dart
// lib/models/payment_request.dart

import 'dart:convert';

class PaymentRequest {
  final int studentId;
  final int? feeId; // Optional: if payment is for a specific fee
  final double amount;
  final String currency;
  final String description;

  PaymentRequest({
    required this.studentId,
    this.feeId,
    required this.amount,
    required this.currency,
    required this.description,
  });

  // Convert a PaymentRequest object to a JSON string
  String toJson() {
    return jsonEncode({
      'studentId': studentId,
      'feeId': feeId,
      'amount': amount,
      'currency': currency,
      'description': description,
    });
  }
}
